csource/common.go

```diff
@@ -106,6 +114,79 @@ static void install_segv_handler()
 	sigaction(SIGBUS, &sa, NULL);
 }
 
+#ifdef __NR_syz_emit_ethernet
+#define COMMAND_MAX_LEN 64
+
+static void execute_command(const char *format, ...)
+{
+	va_list args;
+	int rv;
+	char command[COMMAND_MAX_LEN];
+
+	va_start(args, format);
+
+	rv = vsnprintf(&command[0], sizeof(command), format, args);
+	if (rv < 0) {
+		perror("snprintf()");
+		exitf("tun: snprintf failed");
+	}
+	if ((unsigned)rv >= sizeof(command)) {
+		fail("tun: command \"%s\"doesn't fit into buffer", &command[0]);
+	}
+	if (system(&command[0]) < 0) {
+		perror(&command[0]);
+		exitf("tun: command \"%s\" failed", &command[0]);
+	}
+
+	va_end(args);
+}
+
+int tunfd;
+
+#define LOCAL_MAC "42:42:42:42:42:42"
+#define LOCAL_IPV4 "192.168.1.5"
+#define LOCAL_IPV6 "fe80::4042:42ff:fe42:4242"
+#define REMOTE_MAC "aa:aa:aa:aa:aa:aa"
+#define REMOTE_IPV4 "192.168.1.6"
+
+static void initialize_tun()
+{
+	int rv;
+
+	if (getuid() != 0)
+		return;
+
+	tunfd = open("/dev/net/tun", O_RDWR);
+	if (tunfd == -1) {
+		perror("open()");
+		exitf("tun: can't open /dev/net/tun");
+	}
+	char name[IFNAMSIZ];
+	rv = snprintf(&name[0], sizeof(name), "syz%d", getpid());
+	if (rv < 0) {
+		perror("snprintf()");
+		exitf("tun: snprintf failed");
+	}
+	assert((unsigned)rv < sizeof(name));
```

Comment:
Assert works only in debug build, don't use it, this is not performance critical.
If you need to check something, check it. If the check is optional, drop it.